File: src/web/src/app/models/game.ts
export interface GameListItemDto {
  id: string;
  barcode: string;
  title: string;
  description: string;
  platform: string;
  releaseDate: string | null;
  status: string;
  price: number;
}

export interface GamesListResponse {
  items: GameListItemDto[];
  totalCount: number;
}

export interface GameDetailDto {
  id: string;
  barcode: string;
  title: string;
  description: string;
  platform: string;
  releaseDate: string | null;
  status: string;
  price: number;
  /** Base64 from API; send back as-is on update */
  rowVersion: string;
}

export interface CreateGameDto {
  barcode: string;
  title: string;
  description: string;
  platform: string;
  releaseDate: string | null;
  status: string;
  price: number;
}

export interface UpdateGameDto extends CreateGameDto {
  rowVersion: string;
}

export const PLATFORMS = ['PC', 'PS5', 'PS4', 'XBOX_SERIES', 'XBOX_ONE', 'SWITCH'] as const;
export const STATUSES = ['Upcoming', 'Active', 'Discontinued'] as const;

File: src/api/Newton.Domain/Game.cs
namespace Newton.Domain;

public class Game
{
    public Guid Id { get; set; }
    public string Barcode { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Platform Platform { get; set; }
    public DateOnly? ReleaseDate { get; set; }
    public Status Status { get; set; }
    public decimal Price { get; set; }
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }
    public byte[] RowVersion { get; set; } = [];
}
